### ISSUE TITLE
Copy to clipboard - for a URL or a long string, etc

### ISSUE THREAD

**Issue by gabsie** [visual-framework/vf-core#1505]
Hi there, 

Can we please introduce a way to Copy to clipboard pattern? I suppose we can use either a link 'copy' or a copy to clipboard icon and then a message to show to the user that something has happened -
'Copied to clipboard'

This would be valuable I guess for other services and components, anyone who works with various data-relevant IDs. At the moment this is necessary for a long string - a UUID which data wranglers use all the time.

Here is where I need it right now: attached screenshot (in progress), I've used the Carbon design icon as an example.

![Screenshot 2021-04-26 at 15 41 06](https://user-images.githubusercontent.com/19713125/116101665-ecf18d00-a6a5-11eb-8ed2-2ee5006ed1ed.png)

**Comment by khawkins98** [comment]
As always: I'll ask, can we just have a "copy" button instead? I know it will take more space, but it would also be more obvious. 

If we do use an icon, which should we @cindyebi ?

![image](https://user-images.githubusercontent.com/928100/116422683-44c3fb80-a840-11eb-8d71-8f57802234ec.png)

Related links:

- https://www.carbondesignsystem.com/components/code-snippet/usage/

**Comment by khawkins98** [comment]
There's also a related thing here, which is some way to show non-disruptive information that "action has been done".

Using a button might simplify this as we can change the label:

- Before copying: "Copy the thing"
- After: "You've copied it!"

![image](https://user-images.githubusercontent.com/928100/116423432-f82cf000-a840-11eb-88b4-0e653a07a247.png)

We want to avoid using a disruptive page alert:

![image](https://user-images.githubusercontent.com/928100/116423183-bdc35300-a840-11eb-9e46-c332aee1da2f.png)

**Comment by gabsie** [comment]
hey Ken

This is a very secondary use, and definitely not a priority thing on this page, doesn't deserve to be made into a button..
I just see this working in the same way as on the carbondesignsystem, small, unobstructive and just for the wranglers who need it..
![Uploading Screenshot 2021-04-28 at 15.43.52.png…]()

**Comment by khawkins98** [comment]
Just adding this example from github:

![image](https://user-images.githubusercontent.com/928100/133824295-02285b33-0bdd-46c8-8e5c-818fc3327f20.png)

Hover (no tooltip)

![image](https://user-images.githubusercontent.com/928100/133824342-306cdebc-8324-4563-897f-ab0b0df6f0a1.png)

Copied

![image](https://user-images.githubusercontent.com/928100/133824412-1e741544-792c-492c-834b-120c8b471c43.png)

markup

```html
<div class="d-flex flex-items-center col-3">
    <span class="f6 color-text-secondary">Pending Invite</span>
    <clipboard-copy value="https://github.com/embl-communications/science-in-school/invitations" data-copy-feedback="Copied!" aria-label="Copy invite link" class="js-clipboard-copy zeroclipboard-link color-text-secondary Link--onHover" tabindex="0" role="button">
      <svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-paste js-clipboard-clippy-icon d-inline-block mx-1">
    <path fill-rule="evenodd" d="M5.75 1a.75.75 0 00-.75.75v3c0 .414.336.75.75.75h4.5a.75.75 0 00.75-.75v-3a.75.75 0 00-.75-.75h-4.5zm.75 3V2.5h3V4h-3zm-2.874-.467a.75.75 0 00-.752-1.298A1.75 1.75 0 002 3.75v9.5c0 .966.784 1.75 1.75 1.75h8.5A1.75 1.75 0 0014 13.25v-9.5a1.75 1.75 0 00-.874-1.515.75.75 0 10-.752 1.298.25.25 0 01.126.217v9.5a.25.25 0 01-.25.25h-8.5a.25.25 0 01-.25-.25v-9.5a.25.25 0 01.126-.217z"></path>
</svg>
      <svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-check js-clipboard-check-icon mx-1 color-icon-success d-none">
    <path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path>
</svg>
    </clipboard-copy>
  </div>
```

**Comment by cindyebi** [comment]
Perhaps we could use the same icon-button component we are currently developing for the Bookmark/Favourite component for this? @khawkins98

**Comment by khawkins98** [comment]
We don't have the capacity to action this, it is being moved to the backlog #446
